futures-util/tests/select_next_some.rs

```diff
@@ -7,26 +7,23 @@ use futures_test::future::FutureTestExt;
 
 #[test]
 fn futures_unordered() {
+    // Checks that even though `async_tasks` will yield a `None` and return
```

Comment:
Can you add another more directed test for this specific behavior? Due to random polling order, the `select!` in this example only tests this property with some probability.

Reply:
Doesn't the `pending_once` guarantee this?

First poll of the `select!` generated future must poll both futures, `fut` returns `Pending` because of the `pending_once`, `async_tasks` yields `None` so `select_next_some()` returns `Pending`, all futures are pending so the `select!` yields `Pending`.

Reply:
Ah, i missed the `pending_once`. Yes, that's  sufficient, but would you mind making a more targeted test anyways? This feels more like an integration test and it's hard to tell exactly what components are being tested.